db-engine/src/main/java/com/splicemachine/db/iapi/services/io/DataInputUtil.java

```diff
@@ -84,10 +85,19 @@ public static void skipFully(DataInput in, int skippedBytes)
      * @return
      */
     public static boolean shouldReadOldFormat() {
-        return !BaseDataDictionary.READ_NEW_FORMAT;
+        return readOldFormat!= null && readOldFormat.get()!= null && readOldFormat.get()
+                || !BaseDataDictionary.READ_NEW_FORMAT;
     }
 
     public static boolean shouldWriteOldFormat() {
         return !BaseDataDictionary.WRITE_NEW_FORMAT;
     }
+
+    public static void setReadOldFormat(Boolean b) {
```

Comment:
I cannot find, how methods setReadOldFormat, clearReadOldFormat are used. Are they needed?

Reply:
Found in EE :)